src/cli/global/common.rs

```diff
@@ -110,6 +117,60 @@ pub(super) fn channel_name_from_prefix(
         .unwrap_or_else(|_| prefix_package.repodata_record.channel.clone())
 }
 
+/// Solve package records from [`SparseRepoData`] for the given package MatchSpec
+///
+/// # Returns
+///
+/// The package records (with dependencies records) for the given package
+/// MatchSpec
+pub async fn solve_package_records<AsChannel, ChannelIter>(
+    gateway: &Gateway,
+    channels: ChannelIter,
+    specs: Vec<MatchSpec>,
+) -> miette::Result<Vec<RepoDataRecord>>
+where
+    AsChannel: Into<Channel>,
+    ChannelIter: IntoIterator<Item = AsChannel>,
+{
+    // Get the repodata for the specs
+    let repodata = await_in_progress("fetching repodata for environment", |_| async {
+        gateway
+            .query(
+                channels,
+                [Platform::current(), Platform::NoArch],
```

Comment:
Before this PR, the query was done for a given Platform. however, virtual_packages were computed for the current platform. Is that discrepancy expected / normal?

Reply:
The platform must be given because you can install packages on osx for both osx-arm64 and osx-64. The virtual packages cannot be computed for an arbitrary platform though so we are stuck with the ones for the current platform.

Reply:
Thanks @baszalmstra .

If this is the only use case, shouldn't we guess `best_platform` automatically and remove the` --platform` argument from the global commands?

Reply:
If we can get rid of the argument thatd be great! How would you go about detecting the `best_platform`?

Reply:
Not 100% sure TBH 😆 . If the only case to handle is the `osx-arm` to `osx`, we could maybe detect the right error / empty solve when the current platform is `osx-arm`, and try again on the 'backup' platform? would that work? is that the only case to handle?

Reply:
Im not entirely sure if that would work, because Im not sure which situation you could get into..